### DISCUSSION TITLE
Which USB adapter for AP mode?

### DISCUSSION THREAD

**Discussion by szekelyisz** [RaspAP/raspap-webgui#737] [General]
I want to use the built-in wireless interface on the Pi as a client and have another WLAN interface in AP mode. I tried simultaneous STA+AP mode, but didn't work well, see #710.

I also tried two USB dongles, the TP-Link TL-WN722N and version 2 (v1 is out of production) of the famous Edimax EW-7811Un. Both required a patched version of hostapd and/or patched kernel drivers if worked at all, but I'd like to avoid hacking entirely.

So which USB dongle are people using for this purpose? Any tips appreciated!

**Comment by Noschvie** [comment]
Have a look to this: J&J Wifi - Ralink RT5370

**Comment by billz** [comment]
This user [documented his setup](https://www.reddit.com/r/RaspAP/comments/hamo40/wifi_repeater_mode_wlan0_wlan1_config_is/g0vy6fm/) with the TP-Link TL-WN8200ND. The adapter is 'end of life' according to TP-Link, but the driver installation may work for your model.

```
sudo wget http://downloads.fars-robotics.net/wifi-drivers/install-wifi -O /usr/bin/install-wifi
sudo chmod +x /usr/bin/install-wifi
sudo install-wifi
sudo reboot
```

Per the installer: "The script only works for wifi modules using the rtl8188eu, rtl8188fu, rtl8192eu, rtl8812au, rtl8821cu, 8822bu, mt7610 and mt7612 drivers."

The Edimax EW-7811Un (version 1) works out of the box on all RPi models and many other SBCs I've tested with Armbian. No patching of hostapd or kernel drivers required. I have not tested recent production runs.
